File: problems/3002/solution.cpp
class Solution {
public:
    int maximumSetSize(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> s1(nums1.begin(), nums1.end());
        unordered_set<int> s2(nums2.begin(), nums2.end());
        unordered_set<int> all = s1;
        all.insert(s2.begin(), s2.end());
        int half = nums1.size() / 2;
        return min((int)all.size(), min((int)s1.size(), half) + min((int)s2.size(), half));
    }
};

File: problems/3002/solution.ex
defmodule Solution do
  def maximum_set_size(nums1, nums2) do
    s1 = MapSet.new(nums1)
    s2 = MapSet.new(nums2)
    half = div(length(nums1), 2)
    min(MapSet.size(MapSet.union(s1, s2)), min(MapSet.size(s1), half) + min(MapSet.size(s2), half))
  end
end
